lisa/sut_orchestrator/azure/platform_.py

```diff
@@ -1402,3 +1425,53 @@ def _get_deployable_vhd_path(
             wait_copy_blob(blob_client, vhd_path, log)
 
         return full_vhd_path
+
+    def _node_generate_min_capability(
+        self, req_cap: schema.NodeSpace, azure_cap: schema.NodeSpace
+    ) -> schema.NodeSpace:
+        assert azure_cap.features, "azure node has no disk features."
+        if req_cap.features is None or (
+            not req_cap.features.__contains__(DiskType.DISK_EPHEMERAL)
+            and not req_cap.features.__contains__(DiskType.DISK_PREMIUM)
+            and not req_cap.features.__contains__(DiskType.DISK_STANDARD)
+        ):
+            # If no disk type feature is specified, use least
+            # cost available disk from azure_cap
+            min_cost_disk = AzurePlatform._get_min_cost_disk(azure_cap.features)
+        else:
+            min_cost_disk = AzurePlatform._get_min_cost_disk(req_cap.features)
```

Comment:
The min_cap should be found in overlap of req and cap. There may be logic to check it before, but the min_cap should have this check too. In case it's misused in future, or followed by other people.

Reply:
Reviewing the logic I tried to add here : 

The `req_cap` will contain either one disk feature or none. 

Case 1:
It has no disk feature: In this case we set `min_cost_disk ` as minimum size disk available in `azure_cap` and remove other mutually exclusive disk feature that `azure_cap` might support. When `req_cap.generate_min_capability(azure_cap_copy)` is called later, it will return the `min_cost` with  `min_cost_disk` feature present.

Case 2: 
It has one disk : In this case, `min_cost_disk` is set to the feature present in `req_cap`. we then remove any mutually exclusive feature from `azure_cap`. When `req_cap.generate_min_capability(azure_cap_copy)` is called later, it will return an error if `min_cost_disk` is not present in `azure_cap`.

The logic for overlap is already present in `generate_min_capability` function. I didn't want to make the `_get_min_cost_disk` more complex by re-adding it. Please let me know if I have missed any scenarios.

Reply:
Based on offline discussion, I will add a validation to `req_cap` to ensure no conflict between disk features.